samples/client/petstore/powershell_test/src/IO.Swagger/API/PetApi.ps1

```diff
@@ -0,0 +1,164 @@
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $false)]
+        [IO.Swagger.Model.Pet]
+        ${ apiKey }
+    )
+
+    Process {
+        'Calling method: PetApi-deletePet' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.deletePet(
+            ${ petId },
+            ${ apiKey }
+        )
+    }
+}
+
+function PetApi-findPetsByStatus {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ status }
+    )
+
+    Process {
+        'Calling method: PetApi-findPetsByStatus' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.findPetsByStatus(
+            ${ status }
+        )
+    }
+}
+
+function PetApi-findPetsByTags {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ tags }
+    )
+
+    Process {
+        'Calling method: PetApi-findPetsByTags' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.findPetsByTags(
+            ${ tags }
+        )
+    }
+}
+
+function PetApi-getPetById {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+    )
+
+    Process {
+        'Calling method: PetApi-getPetById' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.getPetById(
+            ${ petId }
+        )
+    }
+}
+
+function PetApi-updatePet {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ body }
+    )
+
+    Process {
+        'Calling method: PetApi-updatePet' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.updatePet(
+            ${ body }
+        )
+    }
+}
+
+function PetApi-updatePetWithForm {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $false)]
+        [IO.Swagger.Model.Pet]
+        ${ name }
```

Comment:
`$name` should have type of `string`. In PoweShell, you can use either [accelerator](https://learn-powershell.net/2013/10/10/quick-hits-finding-creating-and-removing-powershell-type-accelerators/) type, such as `[string]` or `[xml]` or .NET type, such as `[System.String]` or `[System.Int64]`.